### PR TITLE
Allowing streams for large uploads to google storage via the GCS client

### PR DESCRIPTION
**Pull request by k0p04ku**
**Testing**
image
image

I was able to successfully stream a video into GS.

### CODE REVIEW

**Review by cross311 — CHANGES_REQUESTED** [review]
I think this is a good change, just want to see testing more concrete that it works.

File: packages/gcp/src/__test__/GcsClient.test.ts [hunk 1]
```diff
@@ -86,5 +115,13 @@ describe("GcsClient", () => {
                 undefined
             );
         });
+
+        it("performs a successful write via readable stream", async () => {
+            const client = gcsClient(config);
+            await client.initialize(ctx);
+            await expect(client.putObjectAsStream(span.context(), mockStream, "fileName")).resolves.toBe(
```
[the remark this answers is not in the document]
Comment (k0p04ku) [reply packages/gcp/src/__test__/GcsClient.test.ts]: I've added an assertion for this now.

**Comment by k0p04ku** [comment]
> I think this is a good change, just want to see testing more concrete that it works.

I've added an assertion regarding what exactly we're reading and writing into GS.

File: packages/gcp/src/__test__/GcsClient.test.ts [hunk 2]
```diff
@@ -34,22 +35,25 @@ describe("GcsClient", () => {
...
+                file: (_) => mockFile,
+            };
+            MockStorage.mockImplementation(() => {
+                return {
+                    bucket: () => mockBucket,
+                };
+            });
         });
 
         it("rejects on error from gcs for put", async () => {
```
Comment (prachi30) [inline packages/gcp/src/__test__/GcsClient.test.ts:59]: nit: can you re word it to be specific to put object?

Comment (prachi30) [reply packages/gcp/src/__test__/GcsClient.test.ts:59]:
```suggestion
        it("rejects on error from gcs for putting objects", async () => {
```

Comment (k0p04ku) [reply packages/gcp/src/__test__/GcsClient.test.ts:59]: Fixed

**Review by prachi30 — APPROVED** [review]

**Review by prachi30 — APPROVED** [review]

File: packages/gcp/src/index.ts [hunk 3]
```diff
@@ -78,6 +79,7 @@ export interface IPubSubMessagePreprocessor {
 
 export interface IGcsClient {
     putObject(spanContext: SpanContext, body: Buffer, key: string): Promise<void>;
+    putObjectAsStream(context: SpanContext, stream: Readable, key: string): Promise<void>;
```
Comment (cross311) [inline packages/gcp/src/index.ts:82]: Mind keeping variable names consistant
```suggestion
    putObjectAsStream(spanContext: SpanContext, stream: Readable, key: string): Promise<void>;
```

**Review by cross311 — CHANGES_REQUESTED** [review]

File: packages/gcp/src/index.ts (hunk 3 above)
Comment (k0p04ku) [reply packages/gcp/src/index.ts]: Fixed.

**Review by cross311 — APPROVED** [review]

**Pull request merged**